lib/Backend/IRType.cpp

```diff
@@ -5,6 +5,9 @@
 
 #include "Backend.h"
 
+#undef MachPtr
+int externMachPtr = MachPtr;
```

Comment:
this feels quite error prone. (both previous state and now)

Can we move all `MachXXX` defs under IRType.h and `ifdef` based on archs?

Reply:
I will happily do so, buy as a more extensive change, I want to be up-front about concern for breakage to the architectures I cannot test on.

Reply:
No worries, CI will test them all